src/fides/api/ops/service/connectors/saas_connector.py

```diff
@@ -148,16 +148,23 @@ def retrieve_data(
         input_data: Dict[str, List[Any]],
     ) -> List[Row]:
         """Retrieve data from SaaS APIs"""
-        # generate initial set of requests if read request is defined, otherwise raise an exception
         self.set_privacy_request_state(privacy_request, node)
-
         query_config: SaaSQueryConfig = self.query_config(node)
-        read_request: Optional[SaaSRequest] = query_config.get_request_by_action("read")
-        delete_request: Optional[SaaSRequest] = query_config.get_request_by_action(
-            "delete"
-        )
 
-        if not read_request:
+        # generate initial set of requests if read request is defined, otherwise raise an exception
+
+        # An endpoint can be defined with multiple 'read' requests if the data for a single
+        # collection can be accessed in multiple ways for example:
+        #
+        # 1) If a collection can be retrieved by using different identities such as email or phone number
+        # 2) The complete set of results for a collection is made up of subsets. For example, to retrieve all tickets
+        #    we must change a 'status' query param from 'active' to 'pending' and finally 'closed'
+        read_requests: List[SaaSRequest] = query_config.get_read_requests_by_identity()
```

Comment:
The `QueryConfig` has the context of the current endpoint and identities so it's a no-param call to get the appropriate read requests